diff -uNr quicktime.orig/jpeg_old.h quicktime/jpeg_old.h
--- quicktime.orig/jpeg_old.h	Mon Oct  1 05:16:38 2001
+++ quicktime/jpeg_old.h	Fri Nov  8 19:02:43 2002
@@ -11,9 +11,9 @@
 #endif
 
 
-#include "jpeg/jpeglib.h"
+#include <jpeglib.h>
 #include <png.h>       /* Need setjmp.h included by png.h */
-#include "pthread.h"
+#include <pthread.h>
 
 struct my_error_mgr {
   struct jpeg_error_mgr pub;	/* "public" fields */
diff -uNr quicktime.orig/libdv.h quicktime/libdv.h
--- quicktime.orig/libdv.h	Mon Dec  3 14:48:32 2001
+++ quicktime/libdv.h	Fri Nov  8 19:02:43 2002
@@ -17,7 +17,7 @@
 #define u_int64_t unsigned long long
 #endif
 
-#include "libdv/dv.h"
+#include <libdv/dv.h>
 #include <pthread.h>
 #include <sys/time.h>
 
--- quicktime4linux-2.3/libmjpeg.h.orig	2008-02-03 14:06:52.000000000 +0100
+++ quicktime4linux-2.3/libmjpeg.h	2008-08-27 23:04:07.574121739 +0200
@@ -30,9 +30,9 @@
 
 
 #include <stdio.h>
-#include "jpeglib.h"
+#include <jpeglib.h>
 #include <png.h>              /* Need setjmp.h as included by png.h */
-#include "pthread.h"
+#include <pthread.h>
 #include <setjmp.h>
 
 #define MAXFIELDS 2
