diff -ur quicktime4linux-1.2/atom.c quicktime4linux-1.2.long/atom.c
--- quicktime4linux-1.2/atom.c	Tue Aug  8 22:06:38 2000
+++ quicktime4linux-1.2.long/atom.c	Sun Nov 19 02:15:27 2000
@@ -95,10 +95,10 @@
 	return (long)result;
 }
 
-long quicktime_atom_read_size64(char *data)
+long long quicktime_atom_read_size64(char *data)
 {
-	unsigned long result;
-	unsigned long a, b, c, d, e, f, g, h;
+	unsigned long long result;
+	unsigned long long a, b, c, d, e, f, g, h;
 	
 	a = (unsigned char)data[0];
 	b = (unsigned char)data[1];
@@ -111,7 +111,7 @@
 
 	result = (a<<56) | (b<<48) | (c<<40) | (d<<32) | (e<<24) | (f<<16) | (g<<8) | h;
 	if(result < HEADER_LENGTH) result = HEADER_LENGTH;
-	return (long)result;
+	return (long long)result;
 }
 
 int quicktime_atom_read_type(char *data, char *type)
diff -ur quicktime4linux-1.2/funcprotos.h quicktime4linux-1.2.long/funcprotos.h
--- quicktime4linux-1.2/funcprotos.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2.long/funcprotos.h	Sun Nov 19 02:15:40 2000
@@ -5,7 +5,7 @@
 
 /* atom handling routines */
 long quicktime_atom_read_size(char *data);
-long quicktime_atom_read_size64(char *data);
+long long quicktime_atom_read_size64(char *data);
 
 quicktime_trak_t* quicktime_add_track(quicktime_moov_t *moov);
 
diff -ur quicktime4linux-1.2/sizes.h quicktime4linux-1.2.long/sizes.h
--- quicktime4linux-1.2/sizes.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2.long/sizes.h	Sun Nov 19 02:22:51 2000
@@ -2,6 +2,6 @@
 #define QUICKTIME_SIZES_H
 
 #define QUICKTIME_INT16 short int
-#define MAX_FILESIZE ((unsigned long)0xff << (sizeof(long) - 13))
+#define MAX_FILESIZE 0x7f000000
 
 #endif
