diff -ur quicktime4linux-1.2.orig/funcprotos.h quicktime4linux-1.2/funcprotos.h
--- quicktime4linux-1.2.orig/funcprotos.h	Sun Nov 19 03:18:18 2000
+++ quicktime4linux-1.2/funcprotos.h	Sun Nov 19 03:27:42 2000
@@ -1,7 +1,7 @@
 #ifndef FUNCPROTOS_H
 #define FUNCPROTOS_H
 
-#include "graphics.h"
+#include <quicktime/graphics.h>
 
 /* atom handling routines */
 long quicktime_atom_read_size(char *data);
diff -ur quicktime4linux-1.2.orig/libdv/dv.h quicktime4linux-1.2/libdv/dv.h
--- quicktime4linux-1.2.orig/libdv/dv.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2/libdv/dv.h	Sun Nov 19 03:23:03 2000
@@ -27,7 +27,7 @@
 #ifndef __DV_H__
 #define __DV_H__
 
-#include <bitstream.h>
+#include <quicktime/libdv/bitstream.h>
 
 #define DV_DCT_248	(1)
 #define DV_DCT_88	(0)
diff -ur quicktime4linux-1.2.orig/libdv/dvprivate.h quicktime4linux-1.2/libdv/dvprivate.h
--- quicktime4linux-1.2.orig/libdv/dvprivate.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2/libdv/dvprivate.h	Sun Nov 19 03:23:26 2000
@@ -1,13 +1,13 @@
 #ifndef DVPRIVATE_H
 #define DVPRIVATE_H
 
-#include "dv.h"
+#include <quicktime/libdv/dv.h>
 #include <glib.h>
 #include <pthread.h>
 #include <sys/time.h>
 
 #ifdef HAVE_FIREWIRE
-#include <libraw1394/raw1394.h>
+#include <quicktime/libraw1394/raw1394.h>
 
 typedef struct
 {
diff -ur quicktime4linux-1.2.orig/libdv/libdv.h quicktime4linux-1.2/libdv/libdv.h
--- quicktime4linux-1.2.orig/libdv/libdv.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2/libdv/libdv.h	Sun Nov 19 03:23:44 2000
@@ -18,7 +18,7 @@
 #define DV_NTSC 0
 #define DV_PAL 1
 
-#include "dvprivate.h"
+#include <quicktime/libdv/dvprivate.h>
 
 // ================================== The frame decoder
 dv_t* dv_new();
diff -ur quicktime4linux-1.2.orig/private.h quicktime4linux-1.2/private.h
--- quicktime4linux-1.2.orig/private.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2/private.h	Sun Nov 19 03:27:27 2000
@@ -6,8 +6,8 @@
 #define HEADER_LENGTH 8
 #define MAXTRACKS 1024
 
-#include "codecs.h"
-#include "sizes.h"
+#include <quicktime/codecs.h>
+#include <quicktime/sizes.h>
 
 typedef struct
 {
diff -ur quicktime4linux-1.2.orig/quicktime.h quicktime4linux-1.2/quicktime.h
--- quicktime4linux-1.2.orig/quicktime.h	Tue Aug  8 22:06:39 2000
+++ quicktime4linux-1.2/quicktime.h	Sun Nov 19 03:27:14 2000
@@ -8,9 +8,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include "private.h"
-#include "funcprotos.h"
-#include "sizes.h"
+#include <quicktime/private.h>
+#include <quicktime/funcprotos.h>
+#include <quicktime/sizes.h>
 
 /* This is the reference for all your library entry points. */
 
