diff -urN aaa/quicktime4linux-2.2/qth264.c quicktime4linux-2.2/qth264.c
--- aaa/quicktime4linux-2.2/qth264.c	2006-05-30 08:57:46.000000000 +0200
+++ quicktime4linux-2.2/qth264.c	2006-09-12 23:48:39.483409960 +0200
@@ -138,7 +138,7 @@
 		x264_param_t default_params;
 		x264_param_default(&default_params);
 // Reset quantizer if fixed bitrate
-		if(codec->param.rc.b_cbr)
+		if(codec->param.rc.i_rc_method == X264_RC_ABR)
 		{
 			codec->param.rc.i_qp_constant = default_params.rc.i_qp_constant;
 			codec->param.rc.i_qp_min = default_params.rc.i_qp_min;
@@ -471,7 +471,7 @@
 		}
 		else
 		if(!strcasecmp(key, "h264_fix_bitrate"))
-			codec->param.rc.b_cbr = (*(int*)value) / 1000;
+			codec->param.rc.i_bitrate = (*(int*)value) / 1000;
 	}
 }
 
