--- quicktime4linux-2.1/mpeg4.c.orig	2006-04-26 17:45:40.000000000 +0200
+++ quicktime4linux-2.1/mpeg4.c	2006-04-26 17:45:45.000000000 +0200
@@ -1035,7 +1035,7 @@
 
         	context->b_quant_factor = 1.25;
         	context->b_quant_offset = 1.25;
-			context->error_resilience = FF_ER_CAREFULL;
+			context->error_resilience = FF_ER_CAREFUL;
 			context->error_concealment = 3;
 			context->frame_skip_cmp = FF_CMP_DCTMAX;
 			context->ildct_cmp = FF_CMP_VSAD;
