--- mpeg4.c~	2008-08-05 10:27:28.000000000 +0200
+++ mpeg4.c	2008-10-04 20:27:28.138564040 +0200
@@ -671,7 +671,7 @@
 
         	context->b_quant_factor = 1.25;
         	context->b_quant_offset = 1.25;
-			context->error_resilience = FF_ER_CAREFUL;
+			context->error_recognition = FF_ER_CAREFUL;
 			context->error_concealment = 3;
 			context->frame_skip_cmp = FF_CMP_DCTMAX;
 			context->ildct_cmp = FF_CMP_VSAD;
--- qtffmpeg.c.org	2008-10-04 20:28:21.528710146 +0200
+++ qtffmpeg.c	2008-10-04 20:28:41.478690493 +0200
@@ -58,7 +58,7 @@
 	printf("    codec_id=%d\n", context->codec_id);
 	printf("    codec_tag=%d\n", context->codec_tag);
 	printf("    workaround_bugs=%d\n", context->workaround_bugs);
-	printf("    error_resilience=%d\n", context->error_resilience);
+	printf("    error_recognition=%d\n", context->error_recognition);
 	printf("    has_b_frames=%d\n", context->has_b_frames);
 	printf("    block_align=%d\n", context->block_align);
 	printf("    parse_only=%d\n", context->parse_only);
@@ -67,7 +67,7 @@
 	printf("    slice_offset=%p\n", context->slice_offset);
 	printf("    error_concealment=%d\n", context->error_concealment);
 	printf("    dsp_mask=%p\n", context->dsp_mask);
-	printf("    bits_per_sample=%d\n", context->bits_per_sample);
+	printf("    bits_per_coded_sample=%d\n", context->bits_per_coded_sample);
 	printf("    slice_flags=%d\n", context->slice_flags);
 	printf("    xvmc_acceleration=%d\n", context->xvmc_acceleration);
 	printf("    antialias_algo=%d\n", context->antialias_algo);
--- wma.c~	2007-01-09 08:09:49.000000000 +0100
+++ wma.c	2008-10-04 20:29:22.391811636 +0200
@@ -187,7 +187,7 @@
 
 // Decode chunk into work buffer.
 		pthread_mutex_lock(&ffmpeg_lock);
-		result = avcodec_decode_audio(codec->decoder_context, 
+		result = avcodec_decode_audio2(codec->decoder_context, 
 			(int16_t*)(codec->work_buffer + codec->output_size * sample_size), 
             &bytes_decoded,
             codec->packet_buffer, 
